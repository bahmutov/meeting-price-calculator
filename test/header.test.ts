import xs from 'xstream';
import { assert } from 'jsverify';
import { VNode } from '@cycle/dom';
import { withTime } from 'cyclejs-test-helpers';
import Header from '../src/components/header';
const htmlLooksLike = require('html-looks-like');
const toHtml = require('snabbdom-to-html');

describe('Header Component', () => {
  let originalTimeout;

  beforeEach(function() {
    originalTimeout = jasmine.DEFAULT_TIMEOUT_INTERVAL;
    jasmine.DEFAULT_TIMEOUT_INTERVAL = 20000;
  });

  afterEach(function() {
    jasmine.DEFAULT_TIMEOUT_INTERVAL = originalTimeout;
  });

  it('should render correctly', () => {
    const expectedHTML = `
      <div class="Header">
        <h1>Meeting price calculator</h1>
      </div>
    `;

    const vdom$ = Header();
    const html$ = vdom$.map(toHtml);
    const expected$ = xs.of(expectedHTML);
    const looksLike = withTime(Time =>
      Time.assertEqual(html$, expected$, htmlLooksLike)
    );
    return assert(looksLike);
  });

  it('should match a snapshot correctly', () => {
    const vdom$: xs<VNode> = Header();
    const html$: xs<{}> = vdom$.map(toHtml);
    expect(html$).toMatchSnapshot();
  });
});
