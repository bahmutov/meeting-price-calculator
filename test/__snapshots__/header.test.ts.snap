// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component should match a snapshot correctly 1`] = `
Stream {
  "_d": false,
  "_dl": Object {},
  "_err": Object {},
  "_ils": Array [],
  "_prod": MapOp {
    "f": [Function],
    "ins": Stream {
      "_d": false,
      "_dl": Object {},
      "_err": Object {},
      "_ils": Array [],
      "_prod": FromArray {
        "a": Array [
          <header.fjvbrko>
            Meeting price calculator
          </header.fjvbrko>,
        ],
        "type": "fromArray",
      },
      "_stopID": Object {},
      "_target": Object {},
    },
    "out": Object {},
    "type": "map",
  },
  "_stopID": Object {},
  "_target": Object {},
}
`;
